app/src/main/java/com/example/candidatecodereview/MainActivity.kt

```diff
@@ -1,11 +1,25 @@
 package com.example.candidatecodereview
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
+
+    private var fragment: MainFragment? = null
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        fragment = MainFragment()
```

Comment:
I would declare `val fragment = MainFragment()` here as opposed to having it as a property at the top of the class. Reasoning that it is only used in this function.

I would also make it immutable and non-nullable (no `?` and `!!` operators when accessing it later in the function).

Also, I would extract everything below `setContentView(R.layout.activity_main)` to a separate private function.